overlord/ifacestate/ifacestate.go

```diff
@@ -638,6 +638,18 @@ func InterfacesRequestsControlHandlerServices(st *state.State) ([]*snap.AppInfo,
 // change for regenerating security profiles, thus returning a change, or do
 // nothing, in which case no change is returned.
 func AdviseReportedSystemKeyMismatch(st *state.State, systemKey any) (*state.Change, error) {
+	var seeded bool
+	err := st.Get("seeded", &seeded)
+	if err != nil && !errors.Is(err, state.ErrNoState) {
+		return nil, err
+	}
+
+	if !seeded {
+		// system not ready yet for checking system-key, clients will need to
+		// wait
+		return nil, errors.New("system not yet seeded")
```

Comment:
Could we return the change of the seeding task here instead?

Reply:
it seems that snap run will not wait in this case but just continue running which is probably not want we want?

Reply:
won't we hit the if else above? don't we get a proper client Error response?

Reply:
> won't we hit the else if above with rspError? don't we get a proper client Error response?

ah right, https://github.com/canonical/snapd/blob/4b2050fc4fbf8d7363c3494bf2ad0ad4bd6bb2e7/cmd/snap/cmd_run.go#L269 but we're still do retries, the only exit path is either exhaust all retries or a success

Reply:
ah damn, you're right, I probably shouldn't be looking at code anymore today